Tools: Tune: Collection of patches to produce ALSA UCM friendly binary control blobs

singalsu opened the pull request:
First commit changes the export format to needed format with additional 8 bytes header that is required UCM's cset-tlv.

The other patches clean up the binary and ASCII text blobs exports for nicer directory structure and file names.

lgirdwood reviewed (approved):
@singalsu next steps are  to
1) prefix all tools with ```sof-``` and make them executable
2) Move tool to appropriate module directory
3) Install them in ```make install`` Cmake rule - @marc-hb  will know what to do here and can help.

lgirdwood commented:
@wszypelt good to merge ? I dont think internal CI is testing these scripts yet.

ShriramShastry reviewed (approved):
I reviewed the changes, and they appear to be appropriate. I have a few suggestions I have added.

Thank you

wszypelt commented:
@lgirdwood the current CI passes correctly

marc-hb commented:
> @lgirdwood the current CI passes correctly

current = https://sof-ci.01.org/sof-pr-viewer/#/build/PR9070/build13904370. These references are lost on every force push, always save them in comments (in my dreams CIs would post that comment...)

singalsu commented on tools/tune/eq/example_fir_eq.m:
```diff
@@ -10,17 +10,19 @@ function example_fir_eq()
 fs = 48e3;
-fn.cpath3 = '../../ctl/ipc3';
-fn.cpath4 = '../../ctl/ipc4';
+fn.cpath3 = '../../ctl/ipc3/eq_fir';
```
I think it's better to leave to another PR. There could be a function dir = sof_directory_create_check(dir);

singalsu commented on tools/tune/eq/example_iir_eq.m:
```diff
@@ -10,17 +10,19 @@ function example_iir_eq()
 fs = 48e3;
-fn.cpath3 = '../../ctl/ipc3';
-fn.cpath4 = '../../ctl/ipc4';
+fn.cpath3 = '../../ctl/ipc3/eq_iir';
```
Yep, maybe leave to next PR since I'm not changing these lines with this topic.

singalsu commented on tools/tune/eq/example_spk_eq.m:
```diff
@@ -15,8 +15,10 @@ function example_spk_eq()
 fs = 48e3;
-fn.cpath3 = '../../ctl/ipc3';
-fn.cpath4 = '../../ctl/ipc4';
+iir.cpath3 = '../../ctl/ipc3/eq_iir';
```
Also better to put into other PR.

singalsu commented on tools/tune/eq/example_spk_eq.m:
```diff
@@ -149,13 +153,15 @@ function example_spk_eq()
 	eq_tplg2_write(fullfile(iir.tpath2, iir.tplg2), bp_iir, 'eq_iir', iir.comment);
 end
 
+rmpath ../common
```
I don't have clear idea how to do it. Also postpone to other PR. We need to also find a way to handle the common functions when the scripts are moved into modules (@lgirdwood).

singalsu commented on tools/tune/multiband_drc/example_multiband_drc.m:
```diff
@@ -140,9 +140,9 @@ function export_multiband_drc(prm)
 tplg_write(tplg1_fn, blob8, "MULTIBAND_DRC");
 tplg2_write(tplg2_fn, blob8_ipc4, "multiband_drc_config", "Exported with script example_multiband_drc.m");
-blob_write(blob3_fn, blob8);
+sof_ucm_blob_write(blob3_fn, blob8);
```
Yep, also into other PR. I'm not resolving these to keep these reminders for changes visible.

singalsu commented:
I updated the Matlab/Octave zeros() function to use uint8 type as @ShriramShastry suggested. I added also the missing license text comment to the new function. There are no other changes.

lgirdwood commented:
@wszypelt good to merge ? I dont think Internal CI is testing matlab atm. Thanks !

marc-hb commented on tools/tune/eq/example_fir_eq.m:
```diff
@@ -21,7 +21,7 @@ function example_fir_eq()
 %% Example 1: Loudness
 %% -------------------
-fn.bin = 'eq_fir_loudness.bin';
+fn.bin = 'eq_fir_loudness.blob';
```
Thanks, I was a bit confused by `.bin`. It's much clearer now with `.blob`!

:-)))

wszypelt commented:
@lgirdwood I've been away for a few days, I'll try this build again. results should be available within an hour
